src/tools/illink/src/ILLink.RoslynAnalyzer/TrimAnalysis/TrimAnalysisVisitor.cs

```diff
@@ -75,16 +75,22 @@ public override void ApplyCondition (FeatureChecksValue featureChecksValue, ref
 		// - 'this' parameter (for annotated methods)
 		// - field reference
 
-		public override MultiValue Visit (IOperation? operation, StateValue argument)
+		public override MultiValue DefaultVisit (IOperation operation, StateValue argument)
 		{
-			var returnValue = base.Visit (operation, argument);
+			var returnValue = base.DefaultVisit (operation, argument);
 
 			// If the return value is empty (TopValue basically) and the Operation tree
 			// reports it as having a constant value, use that as it will automatically cover
 			// cases we don't need/want to handle.
-			if (operation != null && returnValue.IsEmpty () && TryGetConstantValue (operation, out var constValue))
+			if (!returnValue.IsEmpty ())
+				return returnValue;
+
+			if (TryGetConstantValue (operation, out var constValue))
 				return constValue;
 
+			if (operation.Type is not null)
```

Comment:
What does it mean when it's null vs nonnull? A comment here might be helpful if it's not in the doc comments for IOperation.Type.